### PR TITLE
Sync GraphQL schema with Synse-Server:2.1 api changes

### PR DESCRIPTION
**Pull request by Kontazler**

### CODE REVIEW

File: config/emulator/devices/devices.yaml [hunk 1]
```diff
@@ -0,0 +1,105 @@
+version: 1.0
```
Comment (edaniszewski) [inline config/emulator/devices/devices.yaml]: if using the same devices that are present by default with the emulator, then you shouldn't need to specify the devices. these are baked in to the emulator image (at least, the newer ones)

File: compose/dev.yml [hunk 2]
```diff
@@ -8,24 +8,40 @@ services:
...
+    container_name: emulator-plugin
+    image: vaporio/emulator-plugin
+    ports:
+      - 5001:5001
+    volumes:
+      - ../config/emulator/tcp:/tmp/config/
+      - ../config/emulator/devices:/tmp/devices/
+    environment:
+      PLUGIN_CONFIG: /tmp/config
+      PLUGIN_DEVICE_PATH: /tmp/devices
```
Comment (edaniszewski) [inline compose/dev.yml]: env variable here should be PLUGIN_DEVICE_CONFIG for sdk1.0 -- that said, see below: you should be able to use the default emulator devices

File: config/emulator/tcp/config.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+version: 1.0
```
Comment (edaniszewski) [inline config/emulator/tcp/config.yml]: this should also be built-in to the emulator plugin, so you shouldn't need it here unless overriding stuff

File: synse_graphql/prometheus.py [hunk 4]
```diff
@@ -119,7 +94,9 @@ def labels(self):
         ]
 
     def name(self, _type, param):
-        return 'device_{0}_{1}_{2}'.format(self.type, _type, param)
+        return 'device_{}'.format('_'.join([
+            ''.join(char for char in label if char not in punctuation)
+            for label in [self.type, _type, param]]))
```
Comment (edaniszewski) [inline synse_graphql/prometheus.py]: what exactly is this doing?

File: synse_graphql/prometheus.py [hunk 5]
```diff
@@ -160,17 +143,16 @@ class LedDevice(Device):
...
-            if v is None:
-                continue
+        if not self._readings:
+            return
+        for r in self._readings:
             try:
-                self.get_metric('gauge', k).set(self._handlers.get(k)(v))
+                _type, _value = r
+                self.get_metric('gauge', _type).set(
+                    self._handlers.get(_type)(_value))
```
Comment (edaniszewski) [inline synse_graphql/prometheus.py]: this is a bit confusing to me -- could we put in a comment saying what this is doing. Also if `_type` is not in `seld._handlers`, then we'll get `None` back and will be calling `None(_value)` which will fail.

File: synse_graphql/schema/util.py [hunk 6]
```diff
@@ -53,17 +53,16 @@ def make_request(backend, uri):
...
         r = SESSION.get(
             '{}/synse/version'.format(path),
             timeout=config.options.get('timeout'))
-        if r.ok:
-            version = r.json().get('api_version')
-        else:
-            logger.warning('Unable to get API version of Synse.')
-            r.raise_for_status()
+        r.raise_for_status()
+        version = r.json().get('api_version')
```
Comment (edaniszewski) [inline synse_graphql/schema/util.py]: making a note that this requests the version every time. we could do some caching so we don't have to fetch it every time, but that can be done later.

File: synse_graphql/prometheus.py (hunk 4 above)
Comment (Kontazler) [reply synse_graphql/prometheus.py]: I discovered using the sandbox plugin, that prometheus metrics can't have punctuation in their name. Everything failed when `_type='read.ok'`. This just sanitizes the label strings.

Comment (edaniszewski) [reply synse_graphql/prometheus.py]: ah. makes sense now, thanks!

**Pull request merged**